pkg/reconciler/v1alpha1/taskrun/resources/pod.go

```diff
@@ -155,6 +161,39 @@ func makeCredentialInitializer(serviceAccountName, namespace string, kubeclient
 	}, volumes, nil
 }
 
+func makeWorkingDirInitializer(taskSpec v1alpha1.TaskSpec) *corev1.Container {
+	seen := map[string]bool{}
+	script := ""
+	for _, step := range taskSpec.Steps {
+		if step.WorkingDir != "" {
+			p := filepath.Clean(step.WorkingDir)
+			if strings.HasPrefix(p, workspaceDir) && p != workspaceDir && !seen[p] {
+				seen[p] = true
+				command := fmt.Sprintf("mkdir -p %s", p)
+				if script == "" {
+					script = command
+				} else {
+					script = fmt.Sprintf("%s && %s", script, command)
```

Comment:
id love to see a function that _only_ creates the `script` for a list of `workingDirs` with its own unit tests